mcs/build/library.make

```diff
@@ -28,11 +28,23 @@ LIB_MCS_FLAGS += $(patsubst %,-r:%.dll, $(subst =,=$(topdir)/class/lib/$(PROFILE
 
 sourcefile = $(LIBRARY).sources
 
+profile_library = $(PROFILE)_$(LIBRARY)
```

Comment:
Why not simply have `$(PROFILE)_$(LIBRARY_SUBDIR)_$(subst /,-,$(intermediate))_$(LIBRARY)` ? There are going to be some consecutive `_` but that's for generated files so it does not matter.